Configurations.md

```diff
@@ -0,0 +1,1760 @@
+    // body
+}
+
+fn lorem<Ipsum, Dolor, Sit, Amet>(ipsum: Ipsum, dolor: Dolor, sit: Sit, amet: Amet)
+    where Ipsum: IpsumDolorSitAmet, Dolor: DolorSitAmetConsectetur, Sit: SitAmetConsecteturAdipiscing, Amet: AmetConsecteturAdipiscingElit
+{
+    // body
+}
+```
+
+#### `"HorizontalVertical"`:
+
+```rust
+fn lorem<Ipsum, Dolor>(ipsum: Ipsum, dolor: Dolor)
+    where Ipsum: IpsumDolorSitAmet, Dolor: DolorSitAmetConsectetur
+{
+    // body
+}
+
+fn lorem<Ipsum, Dolor, Sit, Amet>(ipsum: Ipsum, dolor: Dolor, sit: Sit, amet: Amet)
+    where Ipsum: IpsumDolorSitAmet,
+          Dolor: DolorSitAmetConsectetur,
+          Sit: SitAmetConsecteturAdipiscing,
+          Amet: AmetConsecteturAdipiscingElit
+{
+    // body
+}
+```
+
+#### `"Mixed"`:
+
+```rust
+fn lorem<Ipsum, Dolor>(ipsum: Ipsum, dolor: Dolor)
+    where Ipsum: IpsumDolorSitAmet, Dolor: DolorSitAmetConsectetur
+{
+    // body
+}
+
+fn lorem<Ipsum, Dolor, Sit, Amet>(ipsum: Ipsum, dolor: Dolor, sit: Sit, amet: Amet)
+    where Ipsum: IpsumDolorSitAmet, Dolor: DolorSitAmetConsectetur,
+          Sit: SitAmetConsecteturAdipiscing, Amet: AmetConsecteturAdipiscingElit
+{
+    // body
+}
+```
+
+#### `"Vertical"`:
+
+```rust
+fn lorem<Ipsum, Dolor>(ipsum: Ipsum, dolor: Dolor)
+    where Ipsum: IpsumDolorSitAmet,
+          Dolor: DolorSitAmetConsectetur
+{
+    // body
+}
+
+fn lorem<Ipsum, Dolor, Sit, Amet>(ipsum: Ipsum, dolor: Dolor, sit: Sit, amet: Amet)
+    where Ipsum: IpsumDolorSitAmet,
+          Dolor: DolorSitAmetConsectetur,
+          Sit: SitAmetConsecteturAdipiscing,
+          Amet: AmetConsecteturAdipiscingElit
+{
+    // body
+}
+```
+
+See also: [`where_density`](#where_density), [`where_pred_indent`](#where_pred_indent), [`where_style`](#where_style).
+
+## `where_pred_indent`
+
+Indentation style of a where predicate
+
+- **Default value**: `"Visual"`
+- **Possible values**: `"Block"`, `"Visual"`
+
+#### `"Block"`:
+
+```rust
+fn lorem<Ipsum, Dolor, Sit, Amet>() -> T
+    where Ipsum: Eq,
+        Dolor: Eq,
+        Sit: Eq,
+        Amet: Eq
+{
+    // body
+}
+```
+
+#### `"Visual"`:
+
+```rust
+fn lorem<Ipsum, Dolor, Sit, Amet>() -> T
+    where Ipsum: Eq,
+          Dolor: Eq,
+          Sit: Eq,
+          Amet: Eq
+{
+    // body
+}
+```
+
+See also: [`where_density`](#where_density), [`where_layout`](#where_layout), [`where_style`](#where_style).
+
+## `where_style`
+
+Overall strategy for where clauses
+
+- **Default value**: `"Default"`
+- **Possible values**: `"Default"`, `"Rfc"`
+
+#### `"Default"`:
+
+```rust
+fn lorem<Ipsum, Dolor, Sit, Amet>() -> T
+    where Ipsum: Eq,
+          Dolor: Eq,
+          Sit: Eq,
+          Amet: Eq
+{
+    // body
+}
+```
+
+#### `"Rfc"`:
+
+```rust
+fn lorem<Ipsum, Dolor, Sit, Amet>() -> T
+where
+    Ipsum: Eq,
+    Dolor: Eq,
+    Sit: Eq,
+    Amet: Eq,
+{
+    // body
+}
+```
+
+See also: [`where_density`](#where_density), [`where_layout`](#where_layout), [`where_pred_indent`](#where_pred_indent).
+
+## `wrap_comments`
+
```

Comment:
MIght be good to warn that this does not work very well and in particular will not remove line breaks